### ISSUE TITLE
Model version labels should be applied only after models are loaded

### ISSUE THREAD

**Issue by JakeTheWise** [tensorflow/serving#1608]
### Describe the problem the feature is intended to solve
When starting a tf model server configured by a config file specifying multiple model versions and version labels, there is currently an issue where the server will try to apply the version labels before the models are actually loaded. This breaks my tf server deployed on kubernetes because every time the node restarts, the server tries to start again from the config file and hits this issue.

### Describe the solution
Make the server wait for all model versions to be loaded before trying to apply version labels.

### Describe alternatives you've considered
So far, I've just been manually fixing the server every time it breaks (bleh). I delete the version labels in the config file, wait for the server to restart/reload the models, then add back the version labels and wait for the server to then add those.

**Comment by JakeTheWise** [comment]
@gowthamkpr any thoughts about how this issue could be resolved?

**Comment by JakeTheWise** [comment]
I just discovered the `--allow_version_labels_for_unavailable_models` option. The docstring is a little unclear: `If true, allows assigning unused version labels to models that are not available yet.` But this may be the solution.

**Comment by JakeTheWise** [comment]
Update: `--allow_version_labels_for_unavailable_models` didn't work.

**Comment by lilao** [comment]
Thanks for reporting the issue. Can you explain in more details why applying version labels before models become available is a problem, and what problems do you see after enabling --allow_version_labels_for_unavailable_models.

**Comment by JakeTheWise** [comment]
I have two model versions. I want to serve them simultaneously, with the labels prod and dev. I specify this in my server config file. But when I spin up the server, I get errors that the server is trying to apply labels to models that are not loaded / unavailable. I'll try to send over the logs.

**Comment by lilao** [comment]
Thanks @JakeTheWise. It seems that allow_version_labels_for_unavailable_models is created specifically for this use case. I can help investigate if you can provide more details (e.g., error message) when allow_version_labels_for_unavailable_models is enabled.

**Comment by gowthamkpr** [comment]
@JakeTheWise Can you please respond to the above comment so we can take the discussion further. Thanks!

**Comment by JakeTheWise** [comment]
Well now I'm actually running into an issue where the server is not ignoring `--model_name` and `--model_base_path` when trying to configure with a `model_config_file`:
`docker run -it tensorflow/serving:2.1.0 --model_config_file gs://sauron-gke-kfp/tf_server_conf.pbtxt` gives me `tensorflow_serving/sources/storage_path/file_system_storage_path_source.cc:362] FileSystemStoragePathSource encountered a filesystem access error: Could not find base path /models/model for servable model`

**Comment by lilao** [comment]
@JakeTheWise The problem you described in https://github.com/tensorflow/serving/issues/1608#issuecomment-629277286 seems to be a different issue. If this is unexpected, can you please open a new issue?

Do you have any updates regarding my question in https://github.com/tensorflow/serving/issues/1608#issuecomment-625881952? Thanks!

**Comment by JakeTheWise** [comment]
allow_version_labels_for_unavailable_models works! I'm just dumb.
